pallets/xcmp-queue/src/lib.rs

```diff
@@ -131,35 +131,125 @@ pub mod pallet {
 			Ok(Some(used.saturating_add(1_000_000)).into())
 		}
 
-		/// Overwrites QueueConfig storage item.
+		/// Overwrites the number of pages of messages which must be in the queue for the other side to be told to
+		/// suspend their sending.
 		///
 		/// - `origin`: Must pass `Root`.
-		/// - other parameters: refer to `QueueConfigData` Struct items
+		/// - `new`: Desired value for `QueueConfigData.suspend_value`
 		///
 		/// - Events:
 		/// - `UpdatedConfig`: On Success.
-		#[pallet::weight(10_000_000 as Weight + T::DbWeight::get().writes(1))]
-		pub fn update_config(
+		#[pallet::weight(10_000_000 as Weight + T::DbWeight::get().reads_writes(1, 1))]
+		pub fn update_suspend_threshold(origin: OriginFor<T>, new: u32) -> DispatchResult {
+			ensure_root(origin)?;
+			let mut current_config: QueueConfigData = <QueueConfig<T>>::get();
+			current_config.suspend_threshold = new;
+
+			<QueueConfig<T>>::put(current_config);
```

Comment:
Why not just use `mutate` instead of the `get` and `put` combo?

Reply:
very true